This is just creating a paper trail on where the data came from and how it is opened.

Note that on 2.6.2020 this underwent a significant revision. Specifically:
* CORD is now releasing metadata.csv, document_parses.tar.gz and cord_19_embeddings.tar.gz all as one one .tar.gz. Thus that is now the starting point for this notebook.
* I began storing the data in a Google Drive folder, thus making the exact version I am working with accessible to all.

The data consists of:
* article metadata (metadata.csv)
* parsed articles (document_parses.tar.gz)
* article embeddings based on the SPECTER model (cord_19_embeddings.tar.gz)

all downloaded from https://www.semanticscholar.org/cord19/download on 19.05.2020.

For those that need to be unzipping and/or re-organization, that is done here.

I also do a quick check on filesizes, but contents of each will be explored in more detail in ExploringDataStructures.ipynb.

__Note__: This just uses command line tools. Unfortunately that means that it is unlikely to port to macOS and almost certainly will not port to Windows. It is just intended to be quick and dirty.

In [ ]:
https://drive.google.com/file/d/17WBtdUOEq77OoOXj62oZP_ysEUyZNKMb/view?usp=sharing

### metadata.csv

In [1]:
!wc -l < metadata.csv

63627


So it contains 63627 lines. It stands to reason that each line contains the metadata of a single article.

In [2]:
!du -h metadata.csv

93M	metadata.csv


It is also 93MB. Which is neither here nor there, but good to keep track of.

### document_parses.tar.gz

In [3]:
!du -h document_parses.tar.gz

1.5G	document_parses.tar.gz


So the tar.gz is 1.5GB.

Now unzip it.

In [4]:
!tar -xzf document_parses.tar.gz

In [5]:
!ls

cord_19_embeddings_5_12.csv  document_parses.tar.gz	    README.md
cord_19_embeddings.tar.gz    ExploringDataStructures.ipynb  Unzipping.ipynb
document_parses		     metadata.csv


Unzipped to a directory document_parses.

In [6]:
!du -h document_parses

4.9G	document_parses/pdf_json
3.5G	document_parses/pmc_json
8.4G	document_parses


At 8.4GB this is quite a but larger but surprisingly manageable for full text.

In [7]:
!ls document_parses/pdf_json/ | wc -l

51868


So that is a bit gnarly, 51868 files in the directory. I guess one per article pdf.

In [8]:
!ls document_parses/pmc_json/ | wc -l

33503


Again a lot of full text parse files. Given that it looks like we have metadata for 63627 article, I guess there is some overlap between the PMC set and the pdf set, or maybe are just missing metadata for some.

### cord_19_embeddings.tar.gz

In [9]:
!du -h cord_19_embeddings.tar.gz

407M	cord_19_embeddings.tar.gz


This tar is just 400MB.

Unzipping it.

In [10]:
!tar -xzf cord_19_embeddings.tar.gz

In [11]:
!ls

cord_19_embeddings_5_12.csv  document_parses.tar.gz	    README.md
cord_19_embeddings.tar.gz    ExploringDataStructures.ipynb  Unzipping.ipynb
document_parses		     metadata.csv


Has unzipped to a file cord_19_embeddings_5_12.csv.

In [12]:
!du -h cord_19_embeddings_5_12.csv

894M	cord_19_embeddings_5_12.csv


About a GB of embeddings data.

In [13]:
!wc -l < cord_19_embeddings_5_12.csv

63571


Similar number of lines (and presumably, records) as metadata.csv. 